content/doc/book/pipeline/syntax.adoc

```diff
@@ -0,0 +1,851 @@
+---
+layout: section
+title: Pipeline Syntax
+---
+:notitle:
+:description:
+:author:
+:email: jenkinsci-docs@googlegroups.com
+:sectanchors:
+:imagesdir: /doc/book/resources
+:toc:
+
+= Pipeline Syntax
+
+This section builds on the information introduced in <<getting-started#,
+Getting Started>>, and should be treated solely as a reference. For more
+information on how to use Pipeline syntax in practical examples, refer to
+<<jenkinsfile#, The Jenkinsfile>> section of this chapter. As of version 2.5 of
+the Pipeline plugin, Pipeline supports two discrete syntaxes which are detailed
+below. For the pros and cons of each, see the <<compare>>.
+
+As discussed in <<getting-started#, Getting Started>>, the most fundamental
+part of a Pipeline is the "step." Basically, steps tell Jenkins _what_ to
+do, and serve as the basic building block for both Declarative and Scripted
+Pipeline syntax.
+
+For an overview of available steps, please refer to the
+link:/doc/pipeline/steps[Pipeline Steps reference]
+which contains a comprehensive list of steps built into Pipeline as well as
+steps provided by plugins.
+
+[role=syntax]
+== Declarative Pipeline
+
+Declarative Pipeline is a relatively recent addition to Jenkins Pipeline
+footnoteref:[declarative-version, Version 2.5 of the "Pipeline plugin"
+introduces support for Declarative Pipeline syntax]
+which presents a more simplified and opinionated syntax on top of the Pipeline
+sub-systems.
+
+All valid Declarative Pipelines must be enclosed within a `pipeline` block, for
+example:
+
+[source,groovy]
+----
+pipeline {
+    /* insert Declarative Pipeline here */
+}
+----
+
+The basic statements and expressions which are valid in Declarative Pipeline
+follow the same rules as
+link:https://groovy-lang.org/syntax.html[Groovy's syntax]
+with the following exceptions:
+
+* The top-level of the Pipeline must be a _block_, specifically: `pipeline { }`
+* No semicolons as statement separators. Each statement has to be on its own
+  line
+* Blocks must only consist of <<declarative-sections>>,
+  <<declarative-directives>>, <<declarative-steps>>, or assignment statements.
+* A property reference statement is treated as no-argument method invocation. So for
+  example, input is treated as input()
+
+
+[[declarative-sections]]
+=== Sections
+
+Sections in Declarative Pipeline typically contain one or more
+<<declarative-directives>> or <<declarative-steps>>.
+
+==== post
+
+The `post` section defines actions which will be run at the end of the Pipeline
+run. A number of additional <<post-conditions>> blocks are supported within the `post` section:
+`always`, `changed`, `failure`, `success`, and `unstable`. These
+blocks allow for the execution of steps at the tail-end of the Pipeline run,
+depending on the status of the Pipeline.
```

Comment:
I eschewed using "completion" because the post section can also be used within a `stage`